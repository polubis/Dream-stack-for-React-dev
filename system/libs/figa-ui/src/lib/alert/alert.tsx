import type { ReactElement } from 'react';
import { Font } from '../font';
import type { AlertProps, AlertType } from './defs';

import c from 'classnames';
import { ErrorIcon, type IconProps, InfoIcon, OkIcon, WarnIcon } from '../icon';

const ICONS_MAP: Record<AlertType, (props: IconProps) => ReactElement> = {
  info: InfoIcon,
  ok: OkIcon,
  error: ErrorIcon,
  warn: WarnIcon,
};

const Alert = ({
  className,
  type = 'info',
  variant = 'filled',
  children,
  fixed,
  maxWidth,
  trimmed,
  style,
  ...props
}: AlertProps) => {
  const Icon = ICONS_MAP[type];

  return (
    <div
      className={c('alert', type, variant, { fixed }, { trimmed }, className)}
      style={{
        ...style,
        maxWidth,
      }}
      {...props}
    >
      <Font variant="b2">{children}</Font>
      <Icon />
    </div>
  );
};

export { Alert };
