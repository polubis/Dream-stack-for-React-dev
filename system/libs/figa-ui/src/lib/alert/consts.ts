const ALERT_VARIANTS = ['outlined', 'filled'] as const;
const ALERT_TYPES = ["error", "warning", "info", "success"] as const;

export { ALERT_VARIANTS, ALERT_TYPES }