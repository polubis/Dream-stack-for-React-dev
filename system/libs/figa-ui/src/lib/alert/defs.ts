import type { ReactNode } from 'react';
import type { ALERT_VARIANTS, ALERT_TYPES } from './consts';

type AlertVariant = (typeof ALERT_VARIANTS)[number];
type AlertType = (typeof ALERT_TYPES)[number];

interface AlertProps {
  className?: string;
  variant?: AlertVariant;
  type?: AlertType;
  trim?: boolean;
  maxWidth?: string;
  children: ReactNode;
  onClose?(): void;
}

export type { AlertProps, AlertVariant, AlertType };
