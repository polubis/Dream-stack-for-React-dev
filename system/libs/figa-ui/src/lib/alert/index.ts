export * from './alert';
export * from './defs';
