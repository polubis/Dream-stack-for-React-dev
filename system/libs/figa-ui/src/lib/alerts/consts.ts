const ALERT_POSITIONS = [
  't-l',
  't-c',
  't-r',
  'c-l',
  'c-c',
  'c-r',
  'b-l',
  'b-c',
  'b-r',
] as const;

export { ALERT_POSITIONS };
