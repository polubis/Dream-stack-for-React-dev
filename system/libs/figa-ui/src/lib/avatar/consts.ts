const AVATAR_SIZES = ['small', 'medium', 'big'] as const;
const AVATAR_SHAPES = ['rounded', 'rectangle'] as const;

export { AVATAR_SHAPES, AVATAR_SIZES };
