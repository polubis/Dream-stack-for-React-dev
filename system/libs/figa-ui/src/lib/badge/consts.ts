const BADGE_VARIANTS = ['filled', 'outlined'] as const;
const BADGE_MOTIVES = ['primary', 'secondary', 'ok'] as const;

export { BADGE_VARIANTS, BADGE_MOTIVES };
