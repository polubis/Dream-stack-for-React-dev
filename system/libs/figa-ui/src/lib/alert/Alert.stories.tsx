import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Alert } from './alert';
import { ALERT_TYPES } from './consts';

export default {
    title: 'Alert',
    component: Alert,
  } as ComponentMeta<typeof Alert>;

  const Template: ComponentStory<typeof Alert> = ({variant}) => {
    return (
    <>
        {ALERT_TYPES.map(type => (
            <div style={{margin: '8px 0'}}>
                <Alert alertType={type} variant={variant} message={`this is a ${type} alert - check it out !`} />
            </div>
        ))}
    </>
    )
}

const EditableTemplate: ComponentStory<typeof Alert> = ({variant, message, alertType}) => {
    return (
        <Alert alertType={alertType} variant={variant} message={message}/>
    )
}

export const Filled = Template.bind({});
Filled.args = {
  variant: 'filled',
};

export const Outlined = Template.bind({});
Outlined.args = {
  variant: 'outlined',
};

export const Editable = EditableTemplate.bind({});
Editable.args = {
    variant: "outlined",
    message: 'add text',
    alertType: "error"
}

Editable.argTypes = {
    alertType: {
      control: { type: "select", options: ["error", "warning", "info", "success"] },
    },
  };
