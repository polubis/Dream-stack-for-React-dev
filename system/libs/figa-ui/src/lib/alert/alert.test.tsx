import { render, screen } from '@testing-library/react';
import { Alert } from './alert';
import { ALERT_TYPES, ALERT_VARIANTS } from './consts';
import { ThemeProvider } from '../theme-provider';

describe('Alert can be used when: ', () => {
  it('[FRAGILE] works with default setup', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <Alert>My text</Alert>
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] assigns classes', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <Alert className="my-class">My text</Alert>
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] assigns classes when fixed and trimmed', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <Alert className="my-class" fixed trimmed>
          My text
        </Alert>
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] assigns type classes', () => {
    ALERT_TYPES.forEach((type) => {
      const { asFragment } = render(
        <ThemeProvider>
          <Alert type={type}>My text</Alert>
        </ThemeProvider>
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('[FRAGILE] assigns variant classes', () => {
    ALERT_VARIANTS.forEach((variant) => {
      const { asFragment } = render(
        <ThemeProvider>
          <Alert variant={variant}>My text</Alert>
        </ThemeProvider>
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
