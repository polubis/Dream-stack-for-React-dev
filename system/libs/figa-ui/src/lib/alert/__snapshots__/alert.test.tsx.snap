// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert can be used when:  [FRAGILE] alert can be closed 1`] = `
<body>
  <div>
    <div
      class="sc-bcXHqe ezgtjF alert info filled my-class"
    >
      <svg
        class="icon info-icon"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 17C12.2833 17 12.521 16.904 12.713 16.712C12.905 16.52 13.0007 16.2827 13 16C13 15.7167 12.904 15.479 12.712 15.287C12.52 15.095 12.2827 14.9993 12 15C11.7167 15 11.479 15.096 11.287 15.288C11.095 15.48 10.9993 15.7173 11 16C11 16.2833 11.096 16.521 11.288 16.713C11.48 16.905 11.7173 17.0007 12 17ZM11 13H13V7H11V13ZM12 22C10.6167 22 9.31667 21.7373 8.1 21.212C6.88333 20.6867 5.825 19.9743 4.925 19.075C4.025 18.175 3.31267 17.1167 2.788 15.9C2.26333 14.6833 2.00067 13.3833 2 12C2 10.6167 2.26267 9.31667 2.788 8.1C3.31333 6.88333 4.02567 5.825 4.925 4.925C5.825 4.025 6.88333 3.31267 8.1 2.788C9.31667 2.26333 10.6167 2.00067 12 2C13.3833 2 14.6833 2.26267 15.9 2.788C17.1167 3.31333 18.175 4.02567 19.075 4.925C19.975 5.825 20.6877 6.88333 21.213 8.1C21.7383 9.31667 22.0007 10.6167 22 12C22 13.3833 21.7373 14.6833 21.212 15.9C20.6867 17.1167 19.9743 18.175 19.075 19.075C18.175 19.975 17.1167 20.6877 15.9 21.213C14.6833 21.7383 13.3833 22.0007 12 22ZM12 20C14.2333 20 16.125 19.225 17.675 17.675C19.225 16.125 20 14.2333 20 12C20 9.76667 19.225 7.875 17.675 6.325C16.125 4.775 14.2333 4 12 4C9.76667 4 7.875 4.775 6.325 6.325C4.775 7.875 4 9.76667 4 12C4 14.2333 4.775 16.125 6.325 17.675C7.875 19.225 9.76667 20 12 20Z"
        />
      </svg>
      <p
        class="font b2 default"
      >
        My text
      </p>
      <button
        aria-label="Close alert"
        class="button size-1 rounded ghost tertiary"
        motive="tertiary"
        shape="rounded"
        size="1"
        variant="ghost"
      >
        <svg
          class="icon close-icon"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.4 19L5 17.6L10.6 12L5 6.4L6.4 5L12 10.6L17.6 5L19 6.4L13.4 12L19 17.6L17.6 19L12 13.4L6.4 19Z"
          />
        </svg>
      </button>
    </div>
  </div>
</body>
`;

exports[`Alert can be used when:  [FRAGILE] assigns classes 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe ezgtjF alert info filled my-class"
  >
    <svg
      class="icon info-icon"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 17C12.2833 17 12.521 16.904 12.713 16.712C12.905 16.52 13.0007 16.2827 13 16C13 15.7167 12.904 15.479 12.712 15.287C12.52 15.095 12.2827 14.9993 12 15C11.7167 15 11.479 15.096 11.287 15.288C11.095 15.48 10.9993 15.7173 11 16C11 16.2833 11.096 16.521 11.288 16.713C11.48 16.905 11.7173 17.0007 12 17ZM11 13H13V7H11V13ZM12 22C10.6167 22 9.31667 21.7373 8.1 21.212C6.88333 20.6867 5.825 19.9743 4.925 19.075C4.025 18.175 3.31267 17.1167 2.788 15.9C2.26333 14.6833 2.00067 13.3833 2 12C2 10.6167 2.26267 9.31667 2.788 8.1C3.31333 6.88333 4.02567 5.825 4.925 4.925C5.825 4.025 6.88333 3.31267 8.1 2.788C9.31667 2.26333 10.6167 2.00067 12 2C13.3833 2 14.6833 2.26267 15.9 2.788C17.1167 3.31333 18.175 4.02567 19.075 4.925C19.975 5.825 20.6877 6.88333 21.213 8.1C21.7383 9.31667 22.0007 10.6167 22 12C22 13.3833 21.7373 14.6833 21.212 15.9C20.6867 17.1167 19.9743 18.175 19.075 19.075C18.175 19.975 17.1167 20.6877 15.9 21.213C14.6833 21.7383 13.3833 22.0007 12 22ZM12 20C14.2333 20 16.125 19.225 17.675 17.675C19.225 16.125 20 14.2333 20 12C20 9.76667 19.225 7.875 17.675 6.325C16.125 4.775 14.2333 4 12 4C9.76667 4 7.875 4.775 6.325 6.325C4.775 7.875 4 9.76667 4 12C4 14.2333 4.775 16.125 6.325 17.675C7.875 19.225 9.76667 20 12 20Z"
      />
    </svg>
    <p
      class="font b2 default"
    >
      My text
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert can be used when:  [FRAGILE] assigns classes when fixed and trimmed 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe ezgtjF alert info filled my-class"
  >
    <svg
      class="icon info-icon"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 17C12.2833 17 12.521 16.904 12.713 16.712C12.905 16.52 13.0007 16.2827 13 16C13 15.7167 12.904 15.479 12.712 15.287C12.52 15.095 12.2827 14.9993 12 15C11.7167 15 11.479 15.096 11.287 15.288C11.095 15.48 10.9993 15.7173 11 16C11 16.2833 11.096 16.521 11.288 16.713C11.48 16.905 11.7173 17.0007 12 17ZM11 13H13V7H11V13ZM12 22C10.6167 22 9.31667 21.7373 8.1 21.212C6.88333 20.6867 5.825 19.9743 4.925 19.075C4.025 18.175 3.31267 17.1167 2.788 15.9C2.26333 14.6833 2.00067 13.3833 2 12C2 10.6167 2.26267 9.31667 2.788 8.1C3.31333 6.88333 4.02567 5.825 4.925 4.925C5.825 4.025 6.88333 3.31267 8.1 2.788C9.31667 2.26333 10.6167 2.00067 12 2C13.3833 2 14.6833 2.26267 15.9 2.788C17.1167 3.31333 18.175 4.02567 19.075 4.925C19.975 5.825 20.6877 6.88333 21.213 8.1C21.7383 9.31667 22.0007 10.6167 22 12C22 13.3833 21.7373 14.6833 21.212 15.9C20.6867 17.1167 19.9743 18.175 19.075 19.075C18.175 19.975 17.1167 20.6877 15.9 21.213C14.6833 21.7383 13.3833 22.0007 12 22ZM12 20C14.2333 20 16.125 19.225 17.675 17.675C19.225 16.125 20 14.2333 20 12C20 9.76667 19.225 7.875 17.675 6.325C16.125 4.775 14.2333 4 12 4C9.76667 4 7.875 4.775 6.325 6.325C4.775 7.875 4 9.76667 4 12C4 14.2333 4.775 16.125 6.325 17.675C7.875 19.225 9.76667 20 12 20Z"
      />
    </svg>
    <p
      class="font b2 default trim"
    >
      My text
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert can be used when:  [FRAGILE] assigns type classes 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe ezgtjF alert ok filled"
  >
    <svg
      class="icon success-icon"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.243 16.3139L6 12.0699L7.414 10.6559L10.243 13.4839L15.899 7.8269L17.314 9.2419L10.243 16.3119V16.3139Z"
      />
      <path
        clip-rule="evenodd"
        d="M1 12C1 5.925 5.925 1 12 1C18.075 1 23 5.925 23 12C23 18.075 18.075 23 12 23C5.925 23 1 18.075 1 12ZM12 21C10.8181 21 9.64778 20.7672 8.55585 20.3149C7.46392 19.8626 6.47177 19.1997 5.63604 18.364C4.80031 17.5282 4.13738 16.5361 3.68508 15.4442C3.23279 14.3522 3 13.1819 3 12C3 10.8181 3.23279 9.64778 3.68508 8.55585C4.13738 7.46392 4.80031 6.47177 5.63604 5.63604C6.47177 4.80031 7.46392 4.13738 8.55585 3.68508C9.64778 3.23279 10.8181 3 12 3C14.3869 3 16.6761 3.94821 18.364 5.63604C20.0518 7.32387 21 9.61305 21 12C21 14.3869 20.0518 16.6761 18.364 18.364C16.6761 20.0518 14.3869 21 12 21Z"
        fill-rule="evenodd"
      />
    </svg>
    <p
      class="font b2 default"
    >
      My text
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert can be used when:  [FRAGILE] assigns type classes 2`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe ezgtjF alert warn filled"
  >
    <svg
      class="icon warn-icon"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.001 10H13.001V15H11.001V10ZM11 16H13V18H11V16Z"
      />
      <path
        d="M13.7679 4.19994C13.4199 3.54494 12.7419 3.13794 11.9999 3.13794C11.2579 3.13794 10.5799 3.54494 10.2319 4.20094L2.89395 18.0639C2.73112 18.3684 2.65046 18.71 2.65993 19.0551C2.6694 19.4002 2.76867 19.7369 2.94795 20.0319C3.12465 20.3283 3.37559 20.5735 3.676 20.7432C3.97641 20.913 4.3159 21.0015 4.66095 20.9999H19.3389C20.0469 20.9999 20.6879 20.6379 21.0529 20.0319C21.2319 19.7368 21.331 19.4001 21.3405 19.0551C21.35 18.71 21.2695 18.3685 21.1069 18.0639L13.7679 4.19994ZM4.66095 18.9999L11.9999 5.13694L19.3439 18.9999H4.66095Z"
      />
    </svg>
    <p
      class="font b2 default"
    >
      My text
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert can be used when:  [FRAGILE] assigns type classes 3`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe ezgtjF alert error filled"
  >
    <svg
      class="icon error-icon"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1.5 12C1.5 14.0767 2.11581 16.1068 3.26957 17.8335C4.42332 19.5602 6.0632 20.906 7.98182 21.7007C9.90045 22.4955 12.0116 22.7034 14.0484 22.2982C16.0852 21.8931 17.9562 20.8931 19.4246 19.4246C20.8931 17.9562 21.8931 16.0852 22.2982 14.0484C22.7034 12.0116 22.4955 9.90045 21.7007 7.98182C20.906 6.0632 19.5602 4.42332 17.8335 3.26957C16.1068 2.11581 14.0767 1.5 12 1.5C9.21523 1.5 6.54451 2.60625 4.57538 4.57538C2.60625 6.54451 1.5 9.21523 1.5 12ZM18.8625 17.8125L6.1875 5.1375C7.91298 3.70125 10.1129 2.96173 12.3556 3.0641C14.5983 3.16646 16.7218 4.10331 18.3092 5.69077C19.8967 7.27824 20.8335 9.40169 20.9359 11.6444C21.0383 13.8871 20.2988 16.087 18.8625 17.8125ZM6.18 18.87C4.36374 17.3276 3.23325 15.1279 3.03642 12.7533C2.83959 10.3787 3.59248 8.02286 5.13 6.2025L17.7975 18.87C16.1732 20.2425 14.1153 20.9955 11.9887 20.9955C9.86219 20.9955 7.80433 20.2425 6.18 18.87Z"
      />
    </svg>
    <p
      class="font b2 default"
    >
      My text
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert can be used when:  [FRAGILE] assigns type classes 4`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe ezgtjF alert info filled"
  >
    <svg
      class="icon info-icon"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 17C12.2833 17 12.521 16.904 12.713 16.712C12.905 16.52 13.0007 16.2827 13 16C13 15.7167 12.904 15.479 12.712 15.287C12.52 15.095 12.2827 14.9993 12 15C11.7167 15 11.479 15.096 11.287 15.288C11.095 15.48 10.9993 15.7173 11 16C11 16.2833 11.096 16.521 11.288 16.713C11.48 16.905 11.7173 17.0007 12 17ZM11 13H13V7H11V13ZM12 22C10.6167 22 9.31667 21.7373 8.1 21.212C6.88333 20.6867 5.825 19.9743 4.925 19.075C4.025 18.175 3.31267 17.1167 2.788 15.9C2.26333 14.6833 2.00067 13.3833 2 12C2 10.6167 2.26267 9.31667 2.788 8.1C3.31333 6.88333 4.02567 5.825 4.925 4.925C5.825 4.025 6.88333 3.31267 8.1 2.788C9.31667 2.26333 10.6167 2.00067 12 2C13.3833 2 14.6833 2.26267 15.9 2.788C17.1167 3.31333 18.175 4.02567 19.075 4.925C19.975 5.825 20.6877 6.88333 21.213 8.1C21.7383 9.31667 22.0007 10.6167 22 12C22 13.3833 21.7373 14.6833 21.212 15.9C20.6867 17.1167 19.9743 18.175 19.075 19.075C18.175 19.975 17.1167 20.6877 15.9 21.213C14.6833 21.7383 13.3833 22.0007 12 22ZM12 20C14.2333 20 16.125 19.225 17.675 17.675C19.225 16.125 20 14.2333 20 12C20 9.76667 19.225 7.875 17.675 6.325C16.125 4.775 14.2333 4 12 4C9.76667 4 7.875 4.775 6.325 6.325C4.775 7.875 4 9.76667 4 12C4 14.2333 4.775 16.125 6.325 17.675C7.875 19.225 9.76667 20 12 20Z"
      />
    </svg>
    <p
      class="font b2 default"
    >
      My text
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert can be used when:  [FRAGILE] assigns variant classes 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe ezgtjF alert info filled"
  >
    <svg
      class="icon info-icon"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 17C12.2833 17 12.521 16.904 12.713 16.712C12.905 16.52 13.0007 16.2827 13 16C13 15.7167 12.904 15.479 12.712 15.287C12.52 15.095 12.2827 14.9993 12 15C11.7167 15 11.479 15.096 11.287 15.288C11.095 15.48 10.9993 15.7173 11 16C11 16.2833 11.096 16.521 11.288 16.713C11.48 16.905 11.7173 17.0007 12 17ZM11 13H13V7H11V13ZM12 22C10.6167 22 9.31667 21.7373 8.1 21.212C6.88333 20.6867 5.825 19.9743 4.925 19.075C4.025 18.175 3.31267 17.1167 2.788 15.9C2.26333 14.6833 2.00067 13.3833 2 12C2 10.6167 2.26267 9.31667 2.788 8.1C3.31333 6.88333 4.02567 5.825 4.925 4.925C5.825 4.025 6.88333 3.31267 8.1 2.788C9.31667 2.26333 10.6167 2.00067 12 2C13.3833 2 14.6833 2.26267 15.9 2.788C17.1167 3.31333 18.175 4.02567 19.075 4.925C19.975 5.825 20.6877 6.88333 21.213 8.1C21.7383 9.31667 22.0007 10.6167 22 12C22 13.3833 21.7373 14.6833 21.212 15.9C20.6867 17.1167 19.9743 18.175 19.075 19.075C18.175 19.975 17.1167 20.6877 15.9 21.213C14.6833 21.7383 13.3833 22.0007 12 22ZM12 20C14.2333 20 16.125 19.225 17.675 17.675C19.225 16.125 20 14.2333 20 12C20 9.76667 19.225 7.875 17.675 6.325C16.125 4.775 14.2333 4 12 4C9.76667 4 7.875 4.775 6.325 6.325C4.775 7.875 4 9.76667 4 12C4 14.2333 4.775 16.125 6.325 17.675C7.875 19.225 9.76667 20 12 20Z"
      />
    </svg>
    <p
      class="font b2 default"
    >
      My text
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert can be used when:  [FRAGILE] assigns variant classes 2`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe ezgtjF alert info outlined"
  >
    <svg
      class="icon info-icon"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 17C12.2833 17 12.521 16.904 12.713 16.712C12.905 16.52 13.0007 16.2827 13 16C13 15.7167 12.904 15.479 12.712 15.287C12.52 15.095 12.2827 14.9993 12 15C11.7167 15 11.479 15.096 11.287 15.288C11.095 15.48 10.9993 15.7173 11 16C11 16.2833 11.096 16.521 11.288 16.713C11.48 16.905 11.7173 17.0007 12 17ZM11 13H13V7H11V13ZM12 22C10.6167 22 9.31667 21.7373 8.1 21.212C6.88333 20.6867 5.825 19.9743 4.925 19.075C4.025 18.175 3.31267 17.1167 2.788 15.9C2.26333 14.6833 2.00067 13.3833 2 12C2 10.6167 2.26267 9.31667 2.788 8.1C3.31333 6.88333 4.02567 5.825 4.925 4.925C5.825 4.025 6.88333 3.31267 8.1 2.788C9.31667 2.26333 10.6167 2.00067 12 2C13.3833 2 14.6833 2.26267 15.9 2.788C17.1167 3.31333 18.175 4.02567 19.075 4.925C19.975 5.825 20.6877 6.88333 21.213 8.1C21.7383 9.31667 22.0007 10.6167 22 12C22 13.3833 21.7373 14.6833 21.212 15.9C20.6867 17.1167 19.9743 18.175 19.075 19.075C18.175 19.975 17.1167 20.6877 15.9 21.213C14.6833 21.7383 13.3833 22.0007 12 22ZM12 20C14.2333 20 16.125 19.225 17.675 17.675C19.225 16.125 20 14.2333 20 12C20 9.76667 19.225 7.875 17.675 6.325C16.125 4.775 14.2333 4 12 4C9.76667 4 7.875 4.775 6.325 6.325C4.775 7.875 4 9.76667 4 12C4 14.2333 4.775 16.125 6.325 17.675C7.875 19.225 9.76667 20 12 20Z"
      />
    </svg>
    <p
      class="font b2 default"
    >
      My text
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert can be used when:  [FRAGILE] works with default setup 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe ezgtjF alert info filled"
  >
    <svg
      class="icon info-icon"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 17C12.2833 17 12.521 16.904 12.713 16.712C12.905 16.52 13.0007 16.2827 13 16C13 15.7167 12.904 15.479 12.712 15.287C12.52 15.095 12.2827 14.9993 12 15C11.7167 15 11.479 15.096 11.287 15.288C11.095 15.48 10.9993 15.7173 11 16C11 16.2833 11.096 16.521 11.288 16.713C11.48 16.905 11.7173 17.0007 12 17ZM11 13H13V7H11V13ZM12 22C10.6167 22 9.31667 21.7373 8.1 21.212C6.88333 20.6867 5.825 19.9743 4.925 19.075C4.025 18.175 3.31267 17.1167 2.788 15.9C2.26333 14.6833 2.00067 13.3833 2 12C2 10.6167 2.26267 9.31667 2.788 8.1C3.31333 6.88333 4.02567 5.825 4.925 4.925C5.825 4.025 6.88333 3.31267 8.1 2.788C9.31667 2.26333 10.6167 2.00067 12 2C13.3833 2 14.6833 2.26267 15.9 2.788C17.1167 3.31333 18.175 4.02567 19.075 4.925C19.975 5.825 20.6877 6.88333 21.213 8.1C21.7383 9.31667 22.0007 10.6167 22 12C22 13.3833 21.7373 14.6833 21.212 15.9C20.6867 17.1167 19.9743 18.175 19.075 19.075C18.175 19.975 17.1167 20.6877 15.9 21.213C14.6833 21.7383 13.3833 22.0007 12 22ZM12 20C14.2333 20 16.125 19.225 17.675 17.675C19.225 16.125 20 14.2333 20 12C20 9.76667 19.225 7.875 17.675 6.325C16.125 4.775 14.2333 4 12 4C9.76667 4 7.875 4.775 6.325 6.325C4.775 7.875 4 9.76667 4 12C4 14.2333 4.775 16.125 6.325 17.675C7.875 19.225 9.76667 20 12 20Z"
      />
    </svg>
    <p
      class="font b2 default"
    >
      My text
    </p>
  </div>
</DocumentFragment>
`;
